src/components/SliceIndexBar.tsx

```diff
@@ -1,40 +1,83 @@
-import { Box, Flex, Center, Text } from "@chakra-ui/react"
+import { Box, Flex, NumberInput, NumberInputField } from "@chakra-ui/react"
+import { StringOrNumber } from "@chakra-ui/utils"
+import { useEffect, useState } from "react"
 
 type SliceIndexBarProps = {
     count: number
     selected: number
     setSlice: (num: number) => void
 }
 const SliceIndexBar = ({ count, selected, setSlice }: SliceIndexBarProps): JSX.Element => {
-    const nElements = Array.from(Array(count).keys())
-    const flex_gap = 0
-    // const [selectedSlice, setSelectedSlice] = React.useState<number>(initial)
-    const handleChange = (ind: number) => {
+    // set input field state seperately from selected slice state and only update slice state when input field is valid
+    const [numInput, setNumInput] = useState<StringOrNumber>(selected)
+    // handleSliceClick is run when you click the slice selector range
+    const handleSliceClick = (ind: number) => {
         setSlice(ind)
+        setNumInput(ind)
     }
+    // group slices
+    const max_steps = count <= 200 ? count : 200 // max_steps maxes out at 200. can change the resolution of steps here if desired (performance trade-off if you go too high)
+    const nSteps = Array.from(Array(max_steps).keys())
+
+    // run handleInputChange when input field is updated
+    const handleInputChange = (input: string) => {
+        setNumInput(input)
+        try {
+            const val = parseInt(input)
+            if (0 <= val && val < count) {
+                setSlice(val)
+            }
+        } catch {
+            // if the string input is not a number within the valid slice range, do nothing.
+        }
+    }
+    // Triggers component re-render when selected is updated
+    useEffect(() => {
+        setNumInput(selected)
+    }, [selected])
+
+    const numDigits = (num: number) => {
+        return Math.max(Math.floor(Math.log10(Math.abs(num))), 0) + 1
+    }
+
+    const max_width = 500 // can make slice selector bar wider by changing this parameter
     return (
-        <Box borderWidth="3px" borderColor="black" maxW="400px" rounded="xl" p="1" boxShadow={"lg"}>
-            <Flex gap={flex_gap}>
-                {nElements.map(function (element, index) {
-                    const color = index <= selected ? "#4299e1" : "lightgrey"
+        <Box
+            borderWidth="3px"
+            borderColor="black"
+            maxW={`${max_width}px`}
+            rounded="lg"
+            p="1"
+            boxShadow="md"
+        >
+            <Flex>
+                {nSteps.map((index) => {
+                    const color = (index * count) / max_steps <= selected ? "blue.300" : "lightgrey"
```

Comment:
```suggestion
                    const color = (index * count) / max_steps <= selected ? "blue.300" : "edf2f7"
```
It as the gray I was changing

Reply:
Oh yeah I see now.